src/sinks/util/encoding/inner.rs

```diff
@@ -0,0 +1,14 @@
+use crate::sinks::util::encoding::TimestampFormat;
+use serde::{Deserialize, Serialize};
+use string_cache::DefaultAtom as Atom;
+
+#[derive(Deserialize, Serialize, Debug, Eq, PartialEq, Clone)]
+pub struct Inner<E> {
+    pub(crate) codec: E,
+    #[serde(default)]
+    pub(crate) only_fields: Option<Vec<Atom>>,
+    #[serde(default)]
+    pub(crate) except_fields: Option<Vec<Atom>>,
+    #[serde(default)]
+    pub(crate) timestamp_format: Option<TimestampFormat>,
+}
```

Comment:
Since these are so tiny and only used by the corresponding config, I recommend moving these into the config sources. If I'm reading right that should eliminate the `pub(crate)` here and drop some `use` clauses.